ota-plus-web/app/com/advancedtelematic/controllers/MetricsController.scala

```diff
@@ -0,0 +1,53 @@
+package com.advancedtelematic.controllers
+
+import java.io.StringWriter
+
+import akka.util.ByteString
+import com.advancedtelematic.metrics.{JvmMetrics, LoggerMetrics, MetricsRepresentation, MetricsSupport}
+import io.prometheus.client.dropwizard.DropwizardExports
+import io.prometheus.client.CollectorRegistry
+import io.prometheus.client.exporter.common.TextFormat
+import javax.inject.{Inject, Singleton}
+import play.api.http.{ContentTypes, Writeable}
+import play.api.mvc.{AbstractController, Action, AnyContent, ControllerComponents}
+
+import scala.concurrent.Await
+
+object MetricsController {
+  import scala.concurrent.ExecutionContext.Implicits.global
+  import scala.concurrent.duration.DurationInt
+  implicit val metricsWriteable: Writeable[MetricsRepresentation] = Writeable( (x: MetricsRepresentation) =>
+    Await.result(x.metricsJson.map(y => ByteString(y.noSpaces)), 0.millis), Some(ContentTypes.JSON))
+}
+
+@Singleton
+class MetricsController @Inject() (components: ControllerComponents) extends AbstractController(components)
+  with MetricsSupport {
+
+  CollectorRegistry.defaultRegistry.register(new DropwizardExports(metricRegistry))
+
+  private[this] val metricSets = List(new JvmMetrics(metricRegistry), new LoggerMetrics(metricRegistry))
+    .map(x => x.urlPrefix -> x).toMap
+
+  def metric(metricSet: String): Action[AnyContent] = Action { _ =>
+    import MetricsController.metricsWriteable
+    metricSets.get(metricSet) match {
+      case Some(ms) => Ok(ms)
+      case None => NotFound
+    }
+  }
+
+  def metricSamples(names: List[String]): Action[AnyContent] = Action { _ =>
+    println(s"Names: ${names}")
```

Comment:
Debugging residue?

Reply:
Just left it there to check if you are really reviewing the PR ;)
good catch, thanks! :)